dask/dataframe/io/csv.py

```diff
@@ -569,8 +571,16 @@ def to_csv(df, filename, name_function=None, compression=None, compute=True,
                        num=df.npartitions, **(storage_options or {}))
 
     to_csv_chunk = delayed(_write_csv, pure=False)
-    values = [to_csv_chunk(d, f, **kwargs)
-              for d, f in zip(df.to_delayed(), files)]
+
+    # If we only want headers in the first file, turn headers off after the
+    # first chunk
+    values = []
```

Comment:
How does this interact with a user-provided `header`?

Hmm I think I would find it cleaner to do something like (untested code)

```python
dfs = df.to_delayed()

values = [to_csv_chunk(dfs[0], files[0], **kwargs)]

if header_first_chunk and len(dfs) > 1:
    kwargs['header'] = False
    values.extend([to_csv_chunk(d, f, **kwargs)
                   for d, f in zip(dfs[1:], files[1:])])
```

Reply:
pandas complains about the header_first_chunk argument since it doesn't know about it, so I remove it on the first iteration